### PR TITLE
Fix issue #36

### PR DESCRIPTION
**Pull request by JoyfulReaper**
I'm not sure why this issue was occurring but using the services instead of the DbContext directly resolves the issue.

### CODE REVIEW

File: CRUDRecipeEF.PL/Program.cs [hunk 1]
```diff
@@ -51,6 +52,7 @@ private static IHostBuilder CreateHostBuilder(string[] args)
               {
                   services.AddTransient<IRecipeService, RecipeService>();
                   services.AddTransient<IIngredientService, IngredientService>();
+                  services.AddTransient<IDataSeed, DataSeed>();
```
Comment (chrisK00) [inline CRUDRecipeEF.PL/Program.cs:55]: When would we want to use this service?

File: CRUDRecipeEF.BL.DL/Data/DataSeed.cs [hunk 2]
```diff
@@ -1,40 +1,54 @@
...
-            context.SaveChanges();
+            _ingredientService = ingredientService;
+            _recipeService = recipeService;
         }
 
+
+        public void Seed()
+        {
+            // This sucks we probably want an recipe factory
+            List<IngredientAddDTO> ingredients = new List<IngredientAddDTO>()
```
Comment (chrisK00) [inline CRUDRecipeEF.BL.DL/Data/DataSeed.cs:25]: A factory is a good idea yea

File: CRUDRecipeEF.BL.DL/Data/DataSeed.cs [hunk 3]
```diff
@@ -1,40 +1,54 @@
...
+            };
+
+            foreach (var ingredient in ingredients)
+            {
+                _ingredientService.AddIngredient(ingredient);
+            }
+
+            _recipeService.AddRecipe(new RecipeAddDTO() { Name = "Chocolate Cake" });
+            _recipeService.AddIngredientToRecipe(ingredients.Where(x => x.Name == "Flour").FirstOrDefault(), "Chocolate Cake");
+            _recipeService.AddIngredientToRecipe(ingredients.Where(x => x.Name == "Chocolate").FirstOrDefault(), "Chocolate Cake");
```
Comment (chrisK00) [inline CRUDRecipeEF.BL.DL/Data/DataSeed.cs]: cant you do firstordefault only without the where?

File: CRUDRecipeEF.BL.DL/Data/DataSeed.cs [hunk 4]
```diff
@@ -1,40 +1,54 @@
...
+using CRUDRecipeEF.BL.DL.DTOs;
+using CRUDRecipeEF.BL.DL.Services;
+using System.Linq;
 
 namespace CRUDRecipeEF.BL.DL.Data
 {
-    public static class DataSeed
+    public class DataSeed : IDataSeed
     {
-        public static void Seed(RecipeContext context)
```
Comment (chrisK00) [inline CRUDRecipeEF.BL.DL/Data/DataSeed.cs:8]: Whats the reason for changing the whole functionality of the class?

File: CRUDRecipeEF.BL.DL/Data/DataSeed.cs [hunk 5]
```diff
@@ -1,40 +1,54 @@
 ﻿using System.Collections.Generic;
-using CRUDRecipeEF.BL.DL.Entities;
+using CRUDRecipeEF.BL.DL.DTOs;
+using CRUDRecipeEF.BL.DL.Services;
+using System.Linq;
 
 namespace CRUDRecipeEF.BL.DL.Data
 {
-    public static class DataSeed
+    public class DataSeed : IDataSeed
```
Comment (chrisK00) [inline CRUDRecipeEF.BL.DL/Data/DataSeed.cs:8]: I tested the seeding yesterday with yannis and i ran it now again. Everything works fine fetching data from data and using services after it has run etc. I think the error is coming from your side

**Review by chrisK00 — CHANGES_REQUESTED** [review]

File: CRUDRecipeEF.BL.DL/Data/DataSeed.cs (hunk 5 above)
Comment (JoyfulReaper) [reply CRUDRecipeEF.BL.DL/Data/DataSeed.cs:8]: How did you test?

File: CRUDRecipeEF.BL.DL/Data/DataSeed.cs (hunk 4 above)
Comment (JoyfulReaper) [reply CRUDRecipeEF.BL.DL/Data/DataSeed.cs:8]: When retrieving recipes they had no ingredients attached even though they were present in the DB

**Comment by JoyfulReaper** [comment]
@chrisK00 found the issue that prevented the original code from working, so this PR is no longer needed.

**Pull request closed without merging**